File: 107598013/src/main/java/entity/Course.java
package entity;

import java.lang.reflect.Field;
import java.util.Objects;

public class Course {
    private Integer id;
    private String title;
    private String description;
    private String suitablePeople;
    private Integer price;
    private String announcement;
    private String remark;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getSuitablePeople() {
        return suitablePeople;
    }

    public void setSuitablePeople(String suitablePeople) {
        this.suitablePeople = suitablePeople;
    }

    public Integer getPrice() {
        return price;
    }

    public void setPrice(Integer price) {
        this.price = price;
    }

    public String getAnnouncement() {
        return announcement;
    }

    public void setAnnouncement(String announcement) {
        this.announcement = announcement;
    }

    public String getRemark() {
        return remark;
    }

    public void setRemark(String remark) {
        this.remark = remark;
    }

    @Override
    public String toString() {
        return String.format("id=%d, Title=%s, Description=%s, SuitablePeople=%s, Price=%d, Announcement=%s, Remark=%s",
                id, title, description, suitablePeople, price, announcement, remark);
    }

    @Override
    public boolean equals(Object object) {
        Field[] fields = getClass().getDeclaredFields();
        for (Field each : fields) {
            if (each.getName().equals("id"))
                continue;

            try {
                if (!Objects.equals(each.get(this), each.get(object))) {
                    return false;
                }
            } catch (IllegalAccessException e) {
                e.printStackTrace();
            }
        }
        return true;
    }
}

File: 107598019/HW2/Software Architecture_XML/Software Architecture/entity/Course.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Software_Architecture.entity
{
    public class Course
    {
        private string title;
        private string description;
        private string suitable;
        private string price;
        private string notice;
        private string other;

        public Course() { }

        public Course(string title, string description, string suitable, string price, string notice, string other) {
            this.title = title;
            this.description = description;
            this.suitable = suitable;
            this.price = price;
            this.notice = notice;
            this.other = other;
        }

        public string getTitle() {
            return title;
        }

        public void setTitle(string title) {
            this.title = title;
        }

        public string getDescription()
        {
            return description;
        }

        public void setDescription(string description)
        {
            this.description = description;
        }

        public string getSuitable()
        {
            return suitable;
        }

        public void setSuitable(string suitable)
        {
            this.suitable = suitable;
        }

        public string getPrice()
        {
            return price;
        }

        public void setPrice(string price)
        {
            this.price = price;
        }

        public string getNotice()
        {
            return notice;
        }

        public void setNotice(string notice)
        {
            this.notice = notice;
        }

        public string getOther()
        {
            return other;
        }

        public void setOther(string other)
        {
            this.other= other;
        }
    }
}
